lib/config.ts

```diff
@@ -0,0 +1,51 @@
+import Common from '@ethereumjs/common'
+import { Logger } from 'winston'
+import { defaultLogger } from './logging'
+
+export interface Options {
+  /**
+   * Specify the chain and hardfork by passing a Common instance.
+   *
+   * If not provided this defaults to chain `mainnet` and hardfork `chainstart`
+   */
+  common?: Common
+  /**
+   * The logger instance with the log level set (winston)
+   */
+  logger?: Logger
+  /**
+   * Synchronization mode ('fast' or 'light')
+   */
+  syncmode?: string
+  /**
+   * Number of peers needed before syncing
+   */
+  minPeers?: number
+  /**
+   * Maximum peers allowed
+   */
+  maxPeers?: number
+}
+
+export class Config {
+  public common: Common
+  public logger: Logger
+  public syncmode: string
+  public minPeers: number
+  public maxPeers: number
+
+  constructor(options: Options = {}) {
+    // Initialize Common with an explicit 'chainstart' HF set until
+    // hardfork awareness is implemented within the library
+    // Also a fix for https://github.com/ethereumjs/ethereumjs-vm/issues/757
+
+    // TODO: map chainParams (and lib/util.parseParams) to new Common format
+    this.common = options.common
+      ? options.common
+      : new Common({ chain: 'mainnet', hardfork: 'chainstart' })
+    this.logger = options.logger ? options.logger : defaultLogger
```

Comment:
I'd suggest to use [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) here.

```ts
this.logger = options.logger ?? defaultLogger
```

Reply:
Oh, that's a nice one, didn't know this operator, will adopt.

Reply:
(generally we should avoid to use too much special syntax (thought this is also not so easy to define what is special and what is common knowledge) to keep the code readable for a broad range of developers, but I would say in this case this is still intuitively graspable even if someone might not know the specific operator)